packages/vue/src/composables/useAuth.ts

```diff
@@ -1,6 +1,14 @@
+import { authMachine } from '@aws-amplify/ui';
 import { useActor } from '@xstate/vue';
-import { authService } from '@aws-amplify/ui';
+import { interpret } from 'xstate';
+import { ref } from 'vue';
+
+const service = ref(
+  interpret(authMachine, {
+    devTools: process.env.NODE_ENV === 'development',
+  }).start()
```

Comment:
I think this will still be a problem because the service is being `start`'d the moment it's imported, rather than created & started _when the Authenticator is rendered_.

Otherwise, it's still a side-effect from an `import` and can break when tree-shaking and be unintended behavior for anyone _not_ using an Authenticator.

Reply:
For comparison, I removed the `authService` for exactly this reason:
https://github.com/aws-amplify/amplify-ui/pull/323/files#diff-06592eff0be23980380da9854605df47ca1355d6d9b5df6fdd9a35aabeb35335L6

Reply:
There might be a way to model React's Context with Vue Provide/Inject, but it will cause `useAuth` to no longer work outside of Vue internally. I've pushed up another possible solution.